src/napari/layers/base/base.py

```diff
@@ -113,6 +117,166 @@ def __call__(self, *args, **kwargs):
         return obj
 
 
+class LayerSlicer(ABC):
```

Comment:
We should rename this as to not confuse it with the existing `napari.components._layer_slicer._LayerSlicer`. `_Slicer` is already fine I think (and should be private).